prettyprinter-configurable/src/Text/PrettyBy/Monad.hs

```diff
@@ -5,40 +5,44 @@
 {-# LANGUAGE RankNTypes             #-}
 
 -- | A monadic interface to configurable pretty-printing.
-
-module Text.PrettyBy.Monad
-    ( HasPrettyConfig (..)
-    , MonadPretty
-    , prettyM
-    , displayM
-    ) where
-
-import Text.Pretty
-import Text.PrettyBy.Default
-import Text.PrettyBy.Internal
-import Text.PrettyBy.Internal.Utils
-
-import Control.Monad.Reader
-import Lens.Micro
+module Text.PrettyBy.Monad (
+  HasPrettyConfig (..),
+  MonadPretty,
+  prettyM,
+  displayM,
+) where
+
+import Control.Monad.Reader (MonadReader)
+import Lens.Micro (Lens')
+import Text.Pretty (Doc)
+import Text.PrettyBy.Default (Render (..))
+import Text.PrettyBy.Internal (PrettyBy (prettyBy))
+import Text.PrettyBy.Internal.Utils (view)
 
 -- | A constraint for \"@config@ is a part of @env@\".
 class HasPrettyConfig env config | env -> config where
-    prettyConfig :: Lens' env config
+  prettyConfig :: Lens' env config
 
 -- @env@ is an artefact of the encoding, it shouldn't be necessary as @m@ determines what it is
 -- and we're not interested in reflecting @env@ explicitly (unlike @config@, which is also
 -- determined by @m@ through @env@, but is useful to have explicitly). But GHC does not like it
 -- when @env@ is left implicit, see https://gitlab.haskell.org/ghc/ghc/issues/3490
+
 -- | A constraint for \"@m@ is a monad that allows to pretty-print values in it by a @config@\".
```

Comment:
Hate. The. Newline.

Like it at least could make it `--\n` instead to connect the comments.

Reply:
One is internal comment, other is a public documentation. I want to be able to quickly and visually distinguish them. Newline is a nice way of separating them. Now I am not going to accidentally think this is a single logical description.